frontend/public/components/factory/list-page.jsx

```diff
@@ -25,66 +25,42 @@ import {
 } from '../utils';
 import { FilterToolbar } from '../filter-toolbar';
 
-/** @type {React.SFC<{disabled?: boolean, label?: string, onChange: React.ChangeEventHandler<any>, defaultValue?: string, value?: string, placeholder?: string, autoFocus?: boolean,}}>} */
-export const TextFilter = ({
-  label,
-  onChange,
-  defaultValue,
-  style,
-  className,
-  value,
-  placeholder = `Filter ${label}...`,
-  autoFocus = false,
-}) => {
-  const input = React.useRef();
-  const onKeyDown = (e) => {
-    const { nodeName } = e.target;
-    if (
-      nodeName === 'INPUT' ||
-      nodeName === 'TEXTAREA' ||
-      e.key !== KEYBOARD_SHORTCUTS.focusFilterInput
-    ) {
-      return;
-    }
-
-    e.stopPropagation();
-    e.preventDefault();
-    input.current.focus();
-  };
-
-  React.useEffect(() => {
-    window.addEventListener('keydown', onKeyDown);
-    // Remove event listeners on cleanup
-    return () => {
-      window.removeEventListener('keydown', onKeyDown);
-    };
-  }, []);
+/** @type {React.SFC<{disabled?: boolean, label?: string, onChange: (value: string) => void;, defaultValue?: string, value?: string, placeholder?: string, autoFocus?: boolean, onFocus?:any, name?:string, id?: string, onKeyDown?: any, parentClassName?: string }}>} */
+export const TextFilter = (props) => {
+  const {
+    label,
+    defaultValue,
+    className,
+    value,
+    placeholder = `Filter ${label}...`,
+    autoFocus = false,
+    onChange,
+    parentClassName,
+  } = props;
+  const { ref } = useDocumentListener();
 
   return (
-    <div className="has-feedback">
-      <input
-        ref={input}
-        autoCapitalize="none"
-        className={classNames('pf-c-form-control co-text-filter', className)}
+    <div className={classNames('has-feedback', parentClassName)}>
+      <TextInput
+        {...props}
+        className={classNames('co-text-filter', className)}
         data-test-id="item-filter"
-        value={value}
-        defaultValue={defaultValue}
-        onChange={onChange}
-        onKeyDown={(e) => e.key === 'Escape' && e.target.blur()}
+        aria-label={placeholder}
         placeholder={placeholder}
-        style={style}
+        onChange={(val) => onChange(val)}
+        ref={ref}
+        autoFocus={autoFocus}
+        defaultValue={defaultValue}
         tabIndex={0}
+        value={value}
         type="text"
-        autoFocus={autoFocus}
-        aria-label={placeholder}
       />
       <span className="form-control-feedback form-control-feedback--keyboard-hint">
         <kbd>/</kbd>
```

Comment:
How do we know the `/` is the shortcut in this context? If `textInputKeyHandler` changes, this could drift from the implementation.

Reply:
Are you implying that we should have the same source for defining the key used by the handler and the key shown here? Sorry I am not following you.

Reply:
> Are you implying that we should have the same source for defining the key used by the handler and the key shown here? Sorry I am not following you.

Yes. Previously it was this component that added the key handler, but that logic is moved outside of the component now. So this component is hard-coding a hint for a shortcut it doesn't manage.